pandas/plotting/_matplotlib/core.py

```diff
@@ -401,6 +401,10 @@ def _compute_plot_data(self):
             include_type = [np.number]
             exclude_type = ["timedelta"]
 
+        # GH 18755, include object and category type for scatter plot
+        if self._kind == "scatter":
+            include_type.extend(["object", "category"])
```

Comment:
Just out of curiosity how is the X axis determined for objects? Is there an implicit sort?

Reply:
For `scatter` plot, i think both x and y axis can plot objects data. btw, what do you mean by `implicit sort`?

i think the the order of objects on X axis will be the same with the order of given objects, e.g. 
```python
df = pd.DataFrame({"x": ["b", "z", "c"], "y": [1, 2, 3], })
plt.scatter(x=df["x"], y=df["y"])
```
on X axis, the order will still be b, z, c, other than b, c, z.  And this is identical to behaviour of matplotlib. Does this answer your question? @WillAyd

Reply:
Could you update or add a test case for mixed object types? Something where the X-axis would be say `["A", 1, "2"]`?

Reply:
emm, but this mixed type is not valid input for scatter plot as far as I know. @WillAyd 

Or do you want to have 
```python
with pytest.raises('something'):
    df.plot(kind='scatter, x='x', y='y')
```

Reply:
No I think I might just misunderstand the PR. So it's not necessarily just `object` dtypes that can be plotted but `object` dtypes containing only numeric values right?

Reply:
yeah, i just did a quick test, and you could see in the picture attached, a `TypeError` will be raised: `TypeError: 1 is not a string`
![Screen Shot 2019-12-24 at 9 49 54 PM](https://user-images.githubusercontent.com/9269816/71424927-83b7cf00-2697-11ea-9d7a-84fddd360c55.png)

Reply:
oops, did not see your comment when commenting 😅 

Yes, the type has to be same on either axis! maybe the name of PR is misleading, sorry about that. I meant `object` type data (all data remains same type) can be plotted now.

Reply:
I'm +/- 0 on the `object` dtype here but maybe its OK if there is an implicit assumption that whatever objects provided are sortable. Can you maybe just start by making your test case `["C", "B", "A"]` along the axis or is that not possible?

Reply:
it is definitely possible, i attach the plot here along with output with matplotlib as reference:
![Screen Shot 2019-12-24 at 10 02 07 PM](https://user-images.githubusercontent.com/9269816/71425064-19a02980-2699-11ea-9357-06c816b0a6d5.png)

just double check a bit: you want such cases to be added to tests, right?